app/generic/proof/facades/acapy_proof_v2.py

```diff
@@ -0,0 +1,159 @@
+            V20PresSendRequestRequest, AdminAPIMessageTracing, V20PresProposalRequest
+        ],
+        free: bool = True,
+        pres_ex_id: str = None,
+    ) -> PresentationExchange:
+        if free:
+            presentation_exchange = (
+                await controller.present_proof_v2_0.send_request_free(
+                    body=proof_request
+                )
+            )
+        elif isinstance(proof_request, AdminAPIMessageTracing) and pres_ex_id:
+            presentation_exchange = await controller.present_proof_v2_0.send_request(
+                pres_ex_id=pres_ex_id, body=proof_request
+            )
+        elif isinstance(proof_request, V20PresProposalRequest):
+            presentation_exchange = await controller.present_proof_v2_0.send_proposal(
+                body=proof_request
+            )
+        else:
+            raise NotImplementedError
+        return cls.__record_to_model(presentation_exchange)
+
+    @classmethod
+    async def accept_proof_request(
+        cls,
+        controller: AcaPyClient,
+        pres_ex_id: str,
+        body: V20PresSpecByFormatRequest,
+    ) -> PresentationExchange:
+        pres_ex_id = cls.__pres_id_no_version(pres_ex_id=pres_ex_id)
+        presentation_record = await controller.present_proof_v2_0.send_presentation(
+            pres_ex_id=pres_ex_id, body=body
+        )
+        return cls.__record_to_model(presentation_record)
+
+    @classmethod
+    async def reject_proof_request(
+        cls,
+        controller: AcaPyClient,
+        pres_ex_id: str,
+        problem_report: str = None,
+    ) -> None:
+        # get the record
+        pres_ex_id = cls.__pres_id_no_version(pres_ex_id=pres_ex_id)
+        proof_request = await controller.present_proof_v2_0.get_record(
+            pres_ex_id=pres_ex_id
+        )
+        # Report problem if desired
+        if problem_report:
+            try:
+                await controller.present_proof_v2_0.report_problem(
+                    pres_ex_id=pres_ex_id,
+                    body=V20PresProblemReportRequest(description=problem_report),
+                )
+            except Exception as e:
+                raise e from e
+        # delete exchange record
+        delete_proof_request_res = await controller.present_proof_v2_0.delete_record(
+            pres_ex_id=pres_ex_id
+        )
+        if not isinstance(proof_request, V20PresExRecord) or not isinstance(
+            delete_proof_request_res, (Dict, NoneType)
+        ):
+            raise HTTPException(status_code=500, detail="Failed to delete record")
+
+    @classmethod
+    def __pres_id_no_version(cls, pres_ex_id: str) -> str:
+        if pres_ex_id.startswith("v1-") or pres_ex_id.startswith("v2-"):
+            return pres_ex_id[3:]
+        else:
+            return pres_ex_id
+
+    @classmethod
+    def __record_to_model(cls, record: V20PresExRecord) -> PresentationExchange:
+
+        return PresentationExchange(
```

Comment:
In the issuer api I added an `protocol_version`  attribute and `IssueCredentialProtocolVersion` enum to indicate the version of the protocol used for the exchange